### PR TITLE
Security review PR - Collection of all "team project" feature related changes

### PR DESCRIPTION
**Pull request by pieterlukasse**
Jira Ticket: [VADC-932](https://ctds-planx.atlassian.net/browse/VADC-932)

### New Features

- Addition of "team project" access control to relevant argo-wrapper endpoints
   - Note: this PR also includes some other changes that happened in between "team project" related PRs (like improvements to CI, changes in Dockerfile, billing id feature) - these can be skipped from review

[VADC-932]: https://ctds-planx.atlassian.net/browse/VADC-932?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/argowrapper/auth/utils.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import requests
+from argowrapper import logger
+
+
+def get_cohort_ids_for_team_project(token, source_id, team_project):
+    header = {"Authorization": token, "cookie": "fence={}".format(token)}
+    url = "http://cohort-middleware-service/cohortdefinition-stats/by-source-id/{}/by-team-project?team-project={}".format(
```
Comment (k-burt-uch) [inline src/argowrapper/auth/utils.py:7]: This should probably be a configurable url - at the very least so no one raises an unnecessary stink about a non-https request grabbing identifiers used for access control logic.

**Comment by k-burt-uch** [comment]
check_auth updated to check teams, and additional annotations were created to check teams across multiple workflows as well as check the cohorts <-> team project association. 

Each endpoint in route has the appropriate annotation to check the team projects authorization depending on whats returned. 

I see no issues with the teams based access control logic used here.

**Review by k-burt-uch — APPROVED** [review]

File: src/argowrapper/auth/utils.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import requests
+from argowrapper import logger
+from argowrapper.constants import (
+    COHORT_DEFINITION_BY_SOURCE_AND_TEAM_PROJECT_URL,
+)
+
+
+def get_cohort_ids_for_team_project(token, source_id, team_project):
+    header = {"Authorization": token, "cookie": "fence={}".format(token)}
+    url1 = "http://cohort-middleware-service/cohortdefinition-stats/by-source-id/{}/by-team-project?team-project={}".format(
```
Comment (k-burt-uch) [inline src/argowrapper/auth/utils.py:10]: @tianj7 is this needed or is this just some leftover cruft that needs to be cleaned up?

**Review by k-burt-uch — APPROVED** [review]

**Review by cmlsn — APPROVED** [review]

File: src/argowrapper/auth/utils.py (hunk 2 above)
Comment (tianj7) [reply src/argowrapper/auth/utils.py]: thanks for catching this, removed.

**Comment by pieterlukasse** [comment]
thanks for the review! Closing this PR and opening https://github.com/uc-cdis/argo-wrapper/pull/125 so the extra changes can be merged to `master`.

**Pull request closed without merging**